### PR TITLE
Create one PyxisGQL instance per thread

### PR DESCRIPTION
**Pull request by qixiang**
GQL's sync transport (`RequestsHTTPTransport`) can't be shared among threads, create one PyxisGQL instance per thread to run Pyxis queries in parallel.

JIRA: CWFHEALTH-1352

### CODE REVIEW

**Comment by qixiang** [comment]
We can use gql's async transport, but it requires much more refactoring work, so at this moment, just create one PyxisGQL instance per thread, in our phase 2 of refactoring work, we can try to use gql's async transport.

**Comment by qixiang** [comment]
Please ignore the CI failures, they have been fixed in the `main` branch.

**Comment by qixiang** [comment]
Forgot to commit a local change to tests/test_image.py, sorry, PR is amended and ready for review.

File: freshmaker/image.py [hunk 1]
```diff
@@ -361,8 +361,13 @@ def resolve_content_sets(self, pyxis_api_instance, children=None):
...
         # Get the published version of this image to find out if the image
         # was actually published.
-        images = pyxis_api_instance.get_images_by_nvrs([self.nvr], published=True)
-        if images:
+        images = pyxis_api_instance.find_images_by_nvr(self.nvr)
+        if not images:
+            log.warning("No image %s found in Pyxis.", self.nvr)
+            return
+
+        published = any([r["published"] is True for img in images for r in img["repositories"]])
```
Comment (ElenaKarolinaSemanova) [inline freshmaker/image.py:369]: Could we omit `is True` ?
```suggestion
        published = any([r["published"] for img in images for r in img["repositories"]])
```

**Review by FernandesMF — APPROVED** [review]
LGTM, I just had minor suggestions. The strategy makes all the sense!

File: freshmaker/image.py (hunk 1 above)
Comment (qixiang) [reply freshmaker/image.py]: You're right, updated.

File: freshmaker/image.py [hunk 2]
```diff
@@ -766,10 +769,15 @@ def get_images_by_nvrs(self, nvrs, published=True, content_sets=None,
         :return: List of containerImages.
```
[the remark this answers is not in the document]
Comment (qixiang) [reply freshmaker/image.py:770]: Done

File: freshmaker/image.py [hunk 3]
```diff
@@ -847,12 +855,15 @@ def find_parent_brew_build_nvr_from_child(self, child_image):
         :rtype: str
```
[the remark this answers is not in the document]
Comment (qixiang) [reply freshmaker/image.py:857]: Done

**Comment by qixiang** [comment]
All comments have been addressed, I'm going to merge this since there is one approval already :)

**Pull request merged**